src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -311,7 +311,24 @@ public SamReader open(final SamInputResource resource) {
                     if (SamStreams.isBAMFile(bufferedStream)) {
```

Comment:
There is no such call on this line so I'm guessing you mean the call just before? Yes, calling toFile on a bucket has no harmful effect, it just returns null. We use other ways to get to the index inside this block, except for the case where there's no index at all (this matches the previous behavior: we pass null and the code deals with the fact there's no index). I've added a comment to this effect.